src/MyConnection.h

```diff
@@ -30,15 +35,21 @@ class MyConnection : boost::noncopyable {
     mysql_options(pConn_, MYSQL_OPT_LOCAL_INFILE, 0);
     // Default to UTF-8
     mysql_options(pConn_, MYSQL_SET_CHARSET_NAME, "UTF8");
-    if (groups != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_GROUP, groups.c_str());
-    if (default_file != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_FILE, default_file.c_str());
-
-
-    if (!mysql_real_connect(pConn_, host.c_str(), user.c_str(),
-        password.c_str(), db == "" ? NULL : db.c_str(), port,
-        unix_socket == "" ? NULL : unix_socket.c_str(), client_flag)) {
+    if (!groups.isNull())
+      mysql_options(pConn_, MYSQL_READ_DEFAULT_GROUP,
+                    Rcpp::as<std::string>(groups).c_str());
```

Comment:
I'm sorry for the time this takes now. With the `CHAR(user[0])` pattern suggested by Hadley, it's a small function again (as there's no copying involved), but I can take care of this later.

I think we can accept this pull request "as is", it would be great if you could open a second pull request that contains this branch plus the "const reference" changes. Thanks again for your help.